rep-I0005.rst

```diff
@@ -0,0 +1,173 @@
+::
+
+   REP: I0005
+   Title: Cartesian Motion Extensions for the Simple Message Protocol
+   Authors: Fred Proctor <proctor@nist.gov> and Tre Harrison <william.harrison@nist.gov>
+   Status: Active
+   Type: Process
+   Content-Type: text/x-rst
+
+Outline
+=======
+
+#. Abstract_
+#. Motivation_
+#. `Message Specifications`_
+#. `Cartesian Move Request`_
+#. `Cartesian Move Reply`_
+#. `Cartesian Move State`_
+#. References_
+#. `Revision History`_
+#. Copyright_
+
+Abstract
+========
+
+This REP documents extensions to the Simple Message protocol
+that support Cartesian motion.
+
+Motivation
+==========
+
+The Simple Mesage protocol defines a low-level interface to robot
+controllers that can easily be implemented. Together with reference
+implementations of Robot Operating System (ROS) trajectory planners,
+Simple Message provides a way for robot vendors to quickly provide
+baseline ROS functionality to their systems. 
+
+Simple Message is a joint-level interface that requires the external
+system to perform the inverse kinematics calculations. Robot
+controllers typically provide this function internally, and the
+proposed extensions enable external systems to benefit from the
+vendor's built-in kinematics. 
+
+Recognizing that robot controllers have limited computing power, data
+representations have been chosen to reduce the burden of converting
+between non-native representations, and reducing the need to to
+convert between different data sizes.
+
+Assumptions
+===========
+
+#. Messages are fixed length, so that robot controllers do not need to
+   dynamically allocate storage for variable length messages.
+#. Integer numbers are 32-bit 2's complement,
+   little-endian. Floating-point numbers are 32-bit little-endian
+   following the IEEE 754 standard. See [#simple_message_tutorial]_.
+#. Units are SI units (meters, radians, seconds).
+#. Message identifiers are unique for each message type.
+#. Sequence numbers associate replies with requests. Replies are send
+   upon receipt of commands, and when they complete, either
+   successfully or with an error. The type of error is not reported.
+#. State feeback messages are sent continuously at a period determined
```

Comment:
Why not send Cartesian pose all the time (specified as `tool0` relative to `base`, see [here](http://wiki.ros.org/Industrial/Tutorials/Create%20a%20URDF%20for%20an%20Industrial%20Robot#Additional.2BAC8-Standard_Frames)).